alchemist/alchemist-swingui/src/main/kotlin/it/unibo/alchemist/input/GenericEvents.kt

```diff
@@ -58,19 +58,19 @@ abstract class AbstractEventDispatcher<T : TriggerAction, E : Event> : EventDisp
 
     protected var triggers: Map<T, (event: E) -> Unit> = emptyMap()
```

Comment:
Do we need protected?

Reply:
Protected is not necessary, but I preferred to create the "set" operator to add triggers instead of making the map public since I did not intend the triggers to be removable once added.

Reply:
Not very clear to me. If it's initialized once and then hidden as an implementation detail, why it's not a `private lateinit`?

Reply:
Now I remember why I made it specifically `protected`. I wanted to allow the inheriting classes to manage the triggers. It's not `lateinit ` for that very reason; this way the triggers can be accessed at any time by the inheriting classes without having to worry whether the map has been initialized or not.

Reply:
OK, then add a documentation comment clearly explaining how subclasses are supposed to inherit it, which is the expected usage.

Thanks

Reply:
It is done